subprojects/language-native/src/main/java/org/gradle/language/nativeplatform/tasks/AbstractNativeCompileTask.java

```diff
@@ -339,11 +351,20 @@ protected FileCollection getHeaderDependencies() throws IOException {
             return null;
         }
 
-        List<String> lines = Files.readLines(inputFile, Charsets.UTF_8);
-        Set<File> files = new HashSet<File>();
-        for (String line : lines) {
-            files.add(new File(line));
-        }
+        List<File> files = Files.readLines(inputFile, Charsets.UTF_8, new LineProcessor<List<File>>() {
```

Comment:
This should be made lazy so that we only read the headerDependenciesFile when we really need it.  Since we'll visit the input properties several times (I think once to validate, once to get dependencies, once to snapshot it, at least), this should be some kind of lazy FileCollection that only reads the contents when `getFiles()` is needed.

Reply:
I made this lazy.